train_jigsaw_evaljig.py

```diff
@@ -136,14 +171,36 @@ def _do_epoch(self):
                 jigsaw_acc = float(jigsaw_correct) / total
                 class_acc = float(class_correct) / total
                 self.logger.log_test(phase, {"jigsaw": jigsaw_acc, "class": class_acc})
-                self.results[phase][self.current_epoch] = class_acc
+                self.results[phase][self.current_epoch] = jigsaw_acc
+
+        ## Save Models
+
+        if (self.results['val'][self.current_epoch] > self.best_jigsaw_acc):
+            self.best_jigsaw_acc = self.results['val'][self.current_epoch]
+            print("Saving new best at epoch: {}".format(self.current_epoch))
+            self.save_model(os.path.join("logs", self.folder_name, 'best_model.pth'))
```

Comment:
Changing to save_folder as it saves the model in the current experiment file, not in it's parent.